### PR TITLE
edit unstructured mesh

### PR DESCRIPTION
**Pull request by hannahsophiadavies**
Alongside adding some code to unstructured mesh I removed the march-native flag from test/cmakelists.txt

### CODE REVIEW

**Review by benbovy — COMMENTED** [review]
Just left a couple of comments below.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -64,7 +64,6 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 # -- Set compiler flags
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR (CMAKE_CXX_COMPILER_ID MATCHES "Intel" AND NOT WIN32))
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native -Wunused-parameter")
```
[the remark this answers is not in the document]
Comment (hannahsophiadavies) [reply test/CMakeLists.txt:67]: The problem was I couldn't compile fastscape on my mac with ARM architecture, removing the -march=native flag fixed it.

**Comment by benbovy** [comment]
Hi Hannah, I've implemented the constructor of the `unstructured_mesh_xt` class (incl. Python bindings) and added a few C++ / Python tests. Node (cell) areas still need to be handled properly. Hope it helps!

**Comment by benbovy** [comment]
@hannahsophiadavies thanks for having uploaded the implementation of of neighbors distances.

In the last commits I've implemented what was still missing (+ fixes, clean-up & tests). I think this PR is complete now, `unstructured_mesh` now implements all grid requirements. There's one test failing on Windows but it is an xtensor issue that (I think?) has been fixed but not released yet. It shouldn't block us from moving forward, though. The Python bindings on windows seem to work well.

Please do not hesitate to give it a try, i.e., run a whole fastscape simulation (channel erosion) using an unstructured mesh (e.g., built using scipy).

**Comment by hannahsophiadavies** [comment]
Ok, I will try a simulation with the UM as soon as I can and let you know!

**Comment by benbovy** [comment]
Let's merge this! Thanks Hannah!

**Pull request merged**